sklearn/linear_model/tests/test_sgd.py

```diff
@@ -467,6 +468,26 @@ def test_set_coef_multiclass(self):
         # Provided intercept_ does match dataset.
         clf = self.factory().fit(X2, Y2, intercept_init=np.zeros((3,)))
 
+    def test_sgd_proba_access(self):
+        # Test for issue #10938
+
+        # Checks if predict_proba and predict_log_proba
+        # is accessible for refrencing before fitting
+        # the SGD classifier
+        clf = SGDClassifier()
+        assert not(hasattr(clf, "predict_proba"))
+        assert not(hasattr(clf, "predict_log_proba"))
+
+        for loss in ["log", "modified_huber"]:
+            clf = SGDClassifier(loss=loss)
+            assert_true(hasattr(clf, "predict_proba"))
+            assert_true(hasattr(clf, "predict_log_proba"))
+
+            # Checks if not fitted check is performed while calling
+            # the methods
+            assert_raises(NotFittedError, clf.predict_proba, [[3, 2]])
```

Comment:
I would expect the NotFittedError case to be already tested in test_common.py so I would remove this from the test.